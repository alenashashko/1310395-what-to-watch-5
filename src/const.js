export const MillisecondsInTimePeriod = {
  SECOND: 1000,
  MINUTE: 60000,
  HOUR: 3600000,
};
