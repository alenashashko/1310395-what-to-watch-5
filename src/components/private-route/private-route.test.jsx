import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {PrivateRoute} from './private-route';
import {AuthorizationStatus, AppRoute} from '../../const';

describe(`Should PrivateRoute render correctly`, () => {
  it(`when user authorised`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <PrivateRoute
              authorizationStatus={AuthorizationStatus.AUTH}
              path={AppRoute.ROOT}
              exact={true}
              render={() => <div>private div</div>}
            />
          </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`when user unauthorised`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <PrivateRoute
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              path={AppRoute.ROOT}
              exact={true}
              render={() => <div>private div</div>}
            />
          </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
