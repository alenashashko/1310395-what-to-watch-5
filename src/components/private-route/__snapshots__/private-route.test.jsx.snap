// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PrivateRoute render correctly with AUTH auth status 1`] = `
<div>
  private div
</div>
`;

exports[`Should PrivateRoute render correctly with NO_AUTH auth status 1`] = `null`;
